internal/webhooks/validate.go

```diff
@@ -0,0 +1,53 @@
+package webhooks
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	addonsv1alpha1 "github.com/openshift/addon-operator/apis/addons/v1alpha1"
+)
+
+func validateInstallSpec(addon addonsv1alpha1.Addon) error {
+	switch addon.Spec.Install.Type {
+	case addonsv1alpha1.OLMOwnNamespace:
+		if addon.Spec.Install.OLMOwnNamespace == nil ||
+			len(addon.Spec.Install.OLMOwnNamespace.Namespace) == 0 {
```

Comment:
@erdii @thetechnick Do we actually need to check if `Namespace` and `CatalogSourceImage` fields are set? 

The CRD specifies both fields as required. The Addon object will be validated against the schema even before it reaches the webhok